### PR TITLE
Create an endpoint, custom exporter and instrumentation for Hilla

### PR DESCRIPTION
**Pull request by MatthewVaadin**
Creates a Hilla endpoint that will process a JSON string. This is called by a custom traces exporter module that takes a reference to the endpoint. The endpoint constructor is then instrumented to inject a callback to do the export from the agent.

### CODE REVIEW

**Review by heruan — CHANGES_REQUESTED** [review]
Still a draft, so this is just a placeholder request to remember to move the Hilla endpoint to its own module.

File: observability-kit-starter/src/main/resources/META-INF/resources/frontend/components/HillaEndpointExporter.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+} from '@opentelemetry/otlp-exporter-base';
+import {
+  createExportTraceServiceRequest,
+  IExportTraceServiceRequest
+} from "@opentelemetry/otlp-transformer";
+
+export class HillaEndpointExporter
+  extends OTLPExporterBrowserBase<ReadableSpan, IExportTraceServiceRequest>
+  implements SpanExporter {
+  protected _endpoint: (string) => void;
```
Comment (Lodin) [inline observability-kit-starter/src/main/resources/META-INF/resources/frontend/components/HillaEndpointExporter.ts:16]: Do I understand correctly that this exporter is designed extendable? 🤔

**Review by Lodin — APPROVED** [review]
LGTM

File: pom.xml [hunk 2]
```diff
@@ -45,6 +46,7 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <flow.version>24.1-SNAPSHOT</flow.version>
+        <hilla.version>2.0.0</hilla.version>
```
Comment (heruan) [inline pom.xml:49]: We should target Hilla 2.1 here to be sure the kit works with the latest platform.
```suggestion
        <hilla.version>2.1-SNAPSHOT</hilla.version>
```

File: observability-kit-starter-hilla/pom.xml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    <parent>
+        <groupId>com.vaadin</groupId>
+        <artifactId>observability-kit</artifactId>
+        <version>2.1-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>observability-kit-starter-hilla</artifactId>
+
+    <properties>
+        <cvdl.name>vaadin-observability-kit</cvdl.name>
```
Comment (heruan) [inline observability-kit-starter-hilla/pom.xml:15]: Let's use `hilla-observability-kit` as the product slug, following the same convention we used for SSO Kit:
```suggestion
        <cvdl.name>hilla-observability-kit</cvdl.name>
```

**Review by heruan — CHANGES_REQUESTED** [review]

**Review by heruan — APPROVED** [review]

**Pull request merged**